attestation/docker/docker_test.go

```diff
@@ -0,0 +1,196 @@
+				"ewogICJidWlsZHguYnVpbGQucmVmIjogInN0cmFuZ2VfbGFsYW5kZS9zdHJhbmdlX2xhbGFuZGUwL2x0eDE2ZTRybnl1MHhwMDc3N29ybWFybXoiLAogICJjb250YWluZXJpbWFnZS5kZXNjcmlwdG9yIjogewogICAgIm1lZGlhVHlwZSI6ICJhcHBsaWNhdGlvbi92bmQub2NpLmltYWdlLmluZGV4LnYxK2pzb24iLAogICAgImRpZ2VzdCI6ICJzaGEyNTY6NGJlZTAzOTY0MWNlMzAwY2IxZDQ2YTVkMThlMDYxMjVhNzBlYjcwYzM1MWVmNjE2YjZlNDlkNzhiN2RlZjU1ZCIsCiAgICAic2l6ZSI6IDE2MDkKICB9LAogICJjb250YWluZXJpbWFnZS5kaWdlc3QiOiAic2hhMjU2OjRiZWUwMzk2NDFjZTMwMGNiMWQ0NmE1ZDE4ZTA2MTI1YTcwZWI3MGMzNTFlZjYxNmI2ZTQ5ZDc4YjdkZWY1NWQiLAogICJpbWFnZS5uYW1lIjogImdoY3IuaW8vY2hhb3NpbnRoZWNyZC9taWMtdGVzdDpsYXRlc3QiCn0=",
+			},
+			validate: func(a *Attestor, err error, name string) {
+				require.Equal(t, "4bee039641ce300cb1d46a5d18e06125a70eb70c351ef616b6e49d78b7def55d", a.ImageDigest[cryptoutil.DigestValue{Hash: crypto.SHA256}], "TestName: %s", name)
+				require.Equal(t, []string{"ghcr.io/chaosinthecrd/mic-test:latest"}, a.ImageReferences, "TestName: %s", name)
+				require.NoError(t, err, "TestName: %s", name)
+			},
+		},
+		{
+			name: "InvalidJsonFile",
+			testProducts: []string{
+				"ewogICJib21Gb3JtYXQiOiAiQ3ljbG9uZURYIiwKICAic3BlY1ZlcnNpb24iOiAiMS40IiwKICAidmVyc2lvbiI6IDEsCiAgIm1ldGFkYXRhIjogewogICAgImNvbXBvbmVudCI6IHsKICAgICAgImJvbS1yZWYiOiAicGtnOmdvbGFuZy9naXRodWIuY29tL2NoYW9zaW50aGVjcmQvbWljLXRlc3RAKGRldmVsKT90eXBlPW1vZHVsZSIsCiAgICAgICJ0eXBlIjogImFwcGxpY2F0aW9uIiwKICAgICAgIm5hbWUiOiAiZ2l0aHViLmNvbS9jaGFvc2ludGhlY3JkL21pYy10ZXN0IiwKICAgICAgInZlcnNpb24iOiAiKGRldmVsKSIsCiAgICAgICJwdXJsIjogInBrZzpnb2xhbmcvZ2l0aHViLmNvbS9jaGFvc2ludGhlY3JkL21pYy10ZXN0QChkZXZlbCk/dHlwZT1tb2R1bGUiLAogICAgICAiZXh0ZXJuYWxSZWZlcmVuY2VzIjogWwogICAgICAgIHsKICAgICAgICAgICJ1cmwiOiAiaHR0cHM6Ly9naXRodWIuY29tL2NoYW9zaW50aGVjcmQvbWljLXRlc3QiLAogICAgICAgICAgInR5cGUiOiAidmNzIgogICAgICAgIH0KICAgICAgXQogICAgfSwKICAgICJwcm9wZXJ0aWVzIjogWwogICAgICB7CiAgICAgICAgIm5hbWUiOiAiY2R4OmdvbW9kOmJpbmFyeTpuYW1lIiwKICAgICAgICAidmFsdWUiOiAib3V0IgogICAgICB9CiAgICBdCiAgfSwKICAiZGVwZW5kZW5jaWVzIjogWwogICAgewogICAgICAicmVmIjogInBrZzpnb2xhbmcvZ2l0aHViLmNvbS9jaGFvc2ludGhlY3JkL21pYy10ZXN0QChkZXZlbCk/dHlwZT1tb2R1bGUiCiAgICB9CiAgXSwKICAiY29tcG9zaXRpb25zIjogWwogICAgewogICAgICAiYWdncmVnYXRlIjogImNvbXBsZXRlIiwKICAgICAgImRlcGVuZGVuY2llcyI6IFsKICAgICAgICAicGtnOmdvbGFuZy9naXRodWIuY29tL2NoYW9zaW50aGVjcmQvbWljLXRlc3RAKGRldmVsKT90eXBlPW1vZHVsZSIKICAgICAgXQogICAgfSwKICAgIHsKICAgICAgImFnZ3JlZ2F0ZSI6ICJ1bmtub3duIgogICAgfQogIF0KfQo=",
+			},
+			validate: func(a *Attestor, err error, name string) {
+				require.Equal(t, a.ImageDigest[cryptoutil.DigestValue{Hash: crypto.SHA256}], "", "TestName: %s", name)
+				require.Equal(t, []string{""}, a.ImageReferences, "TestName: %s", name)
+				require.NoError(t, err, "TestName: %s", name)
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			a := New()
+
+			testProductSet := make(map[string]attestation.Product)
+			for _, prod := range tt.testProducts {
+				decoded, err := base64.StdEncoding.DecodeString(prod)
+				if err != nil {
```

Comment:
Can use `require.NoError` here to be consistent with our other tests. Similar for other places t.Fatal is called.